gnmi/gnmiclient/client.go

```diff
@@ -21,56 +21,51 @@ import (
 	"fmt"
 
 	gpb "github.com/openconfig/gnmi/proto/gnmi"
-	"github.com/openconfig/lemming/gnmi/gnmit"
+	"github.com/openconfig/lemming/gnmi/gnmistore"
 	"github.com/openconfig/ygnmi/ygnmi"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 	"google.golang.org/grpc/credentials/local"
+	"google.golang.org/grpc/metadata"
 )
 
 // localClient is a client that connects to local gNMI cache.
 type localClient struct {
+	gnmiMode gnmistore.GNMIMode
 	gpb.GNMIClient
-	setClient gpb.GNMIClient
 }
 
 // Set uses the datastore client for Set, instead of the public cache endpoint.
 func (m *localClient) Set(ctx context.Context, in *gpb.SetRequest, opts ...grpc.CallOption) (*gpb.SetResponse, error) {
```

Comment:
move this to cacheclient and get rid of localClient entirely?

Reply:
Posted.